tests/test_in_out.py

```diff
@@ -180,3 +180,34 @@ def test_spinner_write_race_condition(capsys):
     assert "aaaa" in out  # spinner text is present
     assert "bbbb" in out  # write() text is present
     assert not re.search(r"aaaa[^\rb]*bbbb", out)
+
+
+def test_spinner_hidden_with_context_manager(capsys):
```

Comment:
This test seems to big for me, trying to check multiple things simultaneously. It may be difficult to pinpoint the exact problem if such test case fails.

Let's split this test case into at least 2 separate ones. One to test basic `hidden` functionality and another one for some advanced stuff (like nesting context managers).

Reply:
I split the test in two tests, one testing the principal spinner and a second one testing the nesting.